contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = add(expenditurePollCounts[structHash], 1);
+      bytes memory claimDelayAction = createClaimDelayAction(poll.action, UINT256_MAX);
+      executeCall(_pollId, claimDelayAction);
+    }
+
+    // Claim tokens if fully staked
+    if (poll.stakes[YAY] == requiredStake && poll.stakes[NAY] == requiredStake) {
+      poll.events[STAKE_END] = now;
+      tokenLocking.claim(token, true);
+    }
+
+    emit PollStaked(_pollId, msg.sender, _vote, amount);
+  }
+
+  function unstakePoll(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    uint256 _vote,
+    uint256 _amount
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Staking, "voting-rep-not-staking");
+    require(now < add(poll.events[CREATE], wmul(stakePeriod, STAKE_COMMIT_FRACTION)), "voting-rep-cannot-withdraw");
+
+    tokenLocking.transfer(token, _amount, msg.sender, true);
+
+    poll.unpaidRewards = sub(poll.unpaidRewards, _amount);
+    poll.stakes[_vote] = sub(poll.stakes[_vote], _amount);
+    stakes[_pollId][msg.sender][_vote] = sub(stakes[_pollId][msg.sender][_vote], _amount);
+
+    emit PollUnstaked(_pollId, msg.sender, _vote, _amount);
+  }
+
+  function submitVote(
+    uint256 _pollId,
+    bytes32 _voteSecret,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Submit, "voting-rep-poll-not-open");
+
+    uint256 userRep = checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    voteSecrets[msg.sender][_pollId] = _voteSecret;
+    poll.repSubmitted = add(poll.repSubmitted, userRep);
+
+    if (poll.repSubmitted >= wmul(poll.skillRep, maxVoteFraction)) {
+      poll.events[SUBMIT_END] = now;
+    }
+
+    emit PollVoteSubmitted(_pollId, msg.sender);
+  }
+
+  function revealVote(
+    uint256 _pollId,
+    bytes32 _salt,
+    uint256 _vote,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Reveal, "voting-rep-poll-not-reveal");
+
+    uint256 userRep = checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    bytes32 voteSecret = voteSecrets[msg.sender][_pollId];
+    require(voteSecret == getVoteSecret(_salt, _vote), "voting-rep-secret-no-match");
+    delete voteSecrets[msg.sender][_pollId];
+
+    poll.votes[_vote] = add(poll.votes[_vote], userRep);
+    poll.repRevealed = add(poll.repRevealed, userRep);
+
+    if (poll.repRevealed >= wmul(poll.skillRep, maxVoteFraction)) {
```

Comment:
This should check `repRevealed` for equality against `repSubmitted` - it's possible not enough submitted to trigger early closure, but who did submit reveals quickly.

Reply:
Works for me.